src/machines/workflow_machines.rs

```diff
@@ -57,6 +59,9 @@ pub(crate) struct WorkflowMachines {
     /// TODO: Make this apply to *all* cancellable things, once we've added more. Key can be enum.
     timer_id_to_machine: HashMap<String, MachineKey>,
 
+    /// TODO document
+    activity_id_to_machine: HashMap<String, MachineKey>,
```

Comment:
We should combine these ID maps by using an enum as the key, mentioned in above TODO

Reply:
Are we sure that activity and timer IDs would never clash?

Reply:
They can't. Enum as key. The key is:

```rust
enum CommandID {
    Timer(String),
    Activity(String)
}
```

Reply:
Added CommandID on top of your recent async changes.

Reply:
Done, same as above.